### PR TITLE
Add `capacity_type` to support node group spot pricing

### PR DESCRIPTION
**Pull request by ChrisMcKee**
## what
* Adds in support for capacity_type settings on node group 
* Removes clrf from document https://www.atechtown.com/remove-ctrl-m-character/
* Adds gitattributes so linux line endings are maintained

## why
* Who doesn't like spot pricing 
* I don't like paying full price for machines

## references
* https://aws.amazon.com/blogs/containers/amazon-eks-now-supports-provisioning-and-managing-ec2-spot-instances-in-managed-node-groups/
* https://aws.amazon.com/about-aws/whats-new/2020/12/amazon-eks-support-ec2-spot-instances-managed-node-groups/
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_node_group#capacity_type

### CODE REVIEW

**Comment by ChrisMcKee** [comment]
image
Tested working.

**Comment by ChrisMcKee** [comment]
one day the readme gods will look kindly on me 😆

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,19 @@
+* text eol=lf
```
Comment (Gowiem) [inline .gitattributes:1]: @ChrisMcKee I'm not sure I fully get why this is needed. It's not in any of our other modules, so I'd like to chat through this a bit before approving. Can you explain this a bit more?

**Review by Gowiem — CHANGES_REQUESTED** [review]

**Comment by Gowiem** [comment]
@ChrisMcKee the README god is real a PITA - don't take it personally 😉 

I forgot that out terratests are currently broken. It's a permissioning issue and should be fixed soon I believe. We'll need to get those fixed and green before we can get this merged. Sorry for the hold up!

File: .gitattributes (hunk 1 above)
Comment (ChrisMcKee) [reply .gitattributes:1]: https://www.git-scm.com/docs/gitattributes 
Its not really a problem that needs fixing so long as no one ever uses windows to edit these projects. 
If you're not on windows the line endings will vary depending on the settings on the machine. I work heavily in .net so subsequently git will try to crlf the line endings. 

Its clean git practice like setting spacing standards etc with editorconfigs and the like. Makes it easier to automate and enforce how your repos maintained 😄

Comment (ChrisMcKee) [reply .gitattributes:1]: Could cut this file down of course to just markdown and tf; I lazily copied it from my geodesic

Comment (ChrisMcKee) [reply .gitattributes:1]: removed all but the wildcard

**Comment by ChrisMcKee** [comment]
Accidentally removed engineering (not sure how 😬 ) fat fingered the tab

**Review by Gowiem — APPROVED** [review]
Looks solid, we'll get this merged once terratests get fixed. 👍

**Review by Gowiem — APPROVED** [review]
LGTM

**Comment by ChrisMcKee** [comment]
Thanks 👍

**Pull request merged**